### PR TITLE
Add final tag for committed mr task attempt

### PR DESCRIPTION
**Pull request by ydai1124**
@ibuenros  can you review?

### CODE REVIEW

File: gobblin-metrics/src/main/java/gobblin/metrics/GobblinMetrics.java [hunk 1]
```diff
@@ -445,6 +446,13 @@ public void stopMetricsReporting() {
     this.metricsReportingStarted = false;
   }
 
+  /**
+   * Stop metric reporting.
+   */
+  public void stopMetricsReporting() {
+    this.stopMetricsReporting(Optional.<Map<String, Object>>absent());
```
Comment (ibuenros) [inline gobblin-metrics/src/main/java/gobblin/metrics/GobblinMetrics.java]: I think an empty map is more appropriate than `Optional.absent()`

**Comment by chavdar** [comment]
Can you check Travis? There are consistent failures in:

    java.lang.AssertionError: expected [true] but found [false] java.lang.AssertionError
    java.lang.AssertionError: expected [true] but found [false]
	at org.testng.Assert.fail(Assert.java:94)
	at org.testng.Assert.failNotEquals(Assert.java:513)
	at org.testng.Assert.assertTrue(Assert.java:42)
	at org.testng.Assert.assertTrue(Assert.java:52)
	at gobblin.metrics.reporter.ScheduledReporterTest.testScheduledReporter(ScheduledReporterTest.java:117)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

And

    java.lang.AssertionError: expected [1] but found [0] java.lang.AssertionError
    java.lang.AssertionError: expected [1] but found [0]
	at org.testng.Assert.fail(Assert.java:94)
	at org.testng.Assert.failNotEquals(Assert.java:513)
	at org.testng.Assert.assertEqualsImpl(Assert.java:135)
	at org.testng.Assert.assertEquals(Assert.java:116)
	at org.testng.Assert.assertEquals(Assert.java:389)
	at org.testng.Assert.assertEquals(Assert.java:399)
	at gobblin.metrics.RootMetricContextTest.testMetricContextLifecycle(RootMetricContextTest.java:156)

**Comment by ydai1124** [comment]
@ibuenros comments addressed. @chavdar build fixed.

File: gobblin-metrics/src/main/java/gobblin/metrics/GobblinMetrics.java [hunk 2]
```diff
@@ -445,6 +447,13 @@ public void stopMetricsReporting() {
     this.metricsReportingStarted = false;
   }
 
+  /**
+   * Stop metric reporting.
+   */
+  public void stopMetricsReporting() {
+    this.stopMetricsReporting(new HashMap<String, Object>());
```
Comment (chavdar) [inline gobblin-metrics/src/main/java/gobblin/metrics/GobblinMetrics.java]: nit: Collections.emptyMap()

File: gobblin-metrics/src/main/java/gobblin/metrics/reporter/ScheduledReporter.java [hunk 3]
```diff
@@ -199,9 +216,26 @@ protected final void report(ReportableContext context) {
...
+    report(context, isFinal, new HashMap<String, Object>());
+  }
+
+
+  /**
+   * @param context {@link InnerMetricContext} to report.
+   * @param isFinal true if this is the final time report will be called for the given context, false otherwise
+   * @see #report(ReportableContext)
+   */
+  protected void report(ReportableContext context, boolean isFinal, Map<String, Object> additionalTags) {
```
Comment (ydai1124) [inline gobblin-metrics/src/main/java/gobblin/metrics/reporter/ScheduledReporter.java]: Not right now.

**Comment by ydai1124** [comment]
@chavdar, this PR restored previous `JobMetrics.stopReporting` logic and added `taskCommitted` event to identify tasks that are comitted when calling `Task.commit()`.

**Pull request merged**